src/main/neighbourcounter/NeighbourCounter.java

```diff
@@ -9,111 +9,33 @@
-    }
-
-    private int getNeighboursToTheNorth(ArrayList<ArrayList<Cell>> list, int xPos, int yPos) {
-        int result = 0;
-        int xDiff = 0;
-        int yDiff = -1;
-
-        if (checkIsInBounds(list, xPos, yPos, xDiff, yDiff)){
-            result += getNeighbour(list, xPos, yPos, xDiff, yDiff);
-        }
-        return result;
-    }
-
-    private int getNeighboursToTheNorthEast(ArrayList<ArrayList<Cell>> list, int xPos, int yPos) {
-        int result = 0;
-        int xDiff = 1;
-        int yDiff = -1;
-
-        if (checkIsInBounds(list, xPos, yPos, xDiff, yDiff)){
-            result += getNeighbour(list, xPos, yPos, xDiff, yDiff);
-        }
-        return result;
-    }
-
-    private int getNeighboursToTheEast(ArrayList<ArrayList<Cell>> list, int xPos, int yPos) {
-        int result = 0;
-        int xDiff = 1;
-        int yDiff = 0;
+    private final int MIN_TEST_RANGE = -1;
+    private final int MAX_TEST_RANGE = 1;
 
-        if (checkIsInBounds(list, xPos, yPos, xDiff, yDiff)){
-            result += getNeighbour(list, xPos, yPos, xDiff, yDiff);
-        }
-        return result;
-    }
-
-    private int getNeighboursToTheSouthWest(ArrayList<ArrayList<Cell>> list, int xPos, int yPos) {
-        int result = 0;
-        int xDiff = -1;
-        int yDiff = 1;
-
-        if (checkIsInBounds(list, xPos, yPos, xDiff, yDiff)){
-            result += getNeighbour(list, xPos, yPos, xDiff, yDiff);
-        }
-        return result;
+    public int count(ArrayList<ArrayList<Cell>> list, int rowNumber, int colNumber) {
+        return processNeighbouringCells(list, rowNumber, colNumber);
     }
 
-    private int getNeighboursToTheSouth(ArrayList<ArrayList<Cell>> list, int xPos, int yPos) {
+    private int processNeighbouringCells(ArrayList<ArrayList<Cell>> list, int rowNumber, int colNumber) {
         int result = 0;
-        int xDiff = 0;
-        int yDiff = 1;
 
-        if (checkIsInBounds(list, xPos, yPos, xDiff, yDiff)){
-            result += getNeighbour(list, xPos, yPos, xDiff, yDiff);
+        for (int rowToTest = MIN_TEST_RANGE; rowToTest <= MAX_TEST_RANGE; rowToTest++) {
+            for (int columnToTest = MIN_TEST_RANGE; columnToTest <= MAX_TEST_RANGE; columnToTest++) {
+                if (isValidNeighbour(list, rowNumber, colNumber, rowToTest, columnToTest)) {
+                    result += getNeighbourCount(list, rowNumber, colNumber, rowToTest, columnToTest);
+                }
+            }
         }
+        
         return result;
     }
 
-    private int getNeighboursToTheSouthEast(ArrayList<ArrayList<Cell>> list, int xPos, int yPos) {
-        int result = 0;
-        int xDiff = 1;
-        int yDiff = 1;
-
-        if (checkIsInBounds(list, xPos, yPos, xDiff, yDiff)){
-            result += getNeighbour(list, xPos, yPos, xDiff, yDiff);
-        }
-        return result;
+    private boolean isValidNeighbour(ArrayList<ArrayList<Cell>> list, int rowNumber, int colNumber, int i, int j) {
```

Comment:
Try don't use variables called "i" or "j".
When you work with variables which always are together you should encapsulate in a Object.
For example Coordinate, Position (row, column)